src/envs/behavior_options.py

```diff
@@ -0,0 +1,976 @@
+def reset_and_release_hand(env) -> None:  # type: ignore
+    """Resets the state of the right hand."""
+    env.robots[0].set_position_orientation(env.robots[0].get_position(),
+                                           env.robots[0].get_orientation())
+    for _ in range(50):
+        env.robots[0].parts["right_hand"].set_close_fraction(0)
+        env.robots[0].parts["right_hand"].trigger_fraction = 0
+        p.stepSimulation()
+
+
+# NOTE: we need a type:ignore because we can't specify the type of env because
+# it would be a igibson.envs.behavior_env.BehaviorEnv, which is an optional
+# import and won't get imported on CI
+def get_delta_low_level_base_action(  # type: ignore
+    env,
+    original_orientation: Tuple,
+    old_xytheta: Array,
+    new_xytheta: Array,
+) -> Array:
+    """Get low-level actions from base movement plan."""
+    ret_action = np.zeros(17, dtype=np.float32)
+
+    robot_z = env.robots[0].get_position()[2]
+
+    # First, get the old and new position and orientation in the world
+    # frame as numpy arrays
+    old_pos = np.array([old_xytheta[0], old_xytheta[1], robot_z])
+    old_orn_quat = p.getQuaternionFromEuler(
+        np.array(
+            [original_orientation[0], original_orientation[1],
+             old_xytheta[2]]))
+    new_pos = np.array([new_xytheta[0], new_xytheta[1], robot_z])
+    new_orn_quat = p.getQuaternionFromEuler(
+        np.array(
+            [original_orientation[0], original_orientation[1],
+             new_xytheta[2]]))
+
+    # Then, simply get the delta position and orientation by multiplying the
+    # inverse of the old pose by the new pose
+    inverted_old_pos, inverted_old_orn_quat = p.invertTransform(
+        old_pos, old_orn_quat)
+    delta_pos, delta_orn_quat = p.multiplyTransforms(inverted_old_pos,
+                                                     inverted_old_orn_quat,
+                                                     new_pos, new_orn_quat)
+
+    # Finally, convert the orientation back to euler angles from a quaternion
+    delta_orn = p.getEulerFromQuaternion(delta_orn_quat)
+
+    ret_action[0:3] = np.array([delta_pos[0], delta_pos[1], delta_orn[2]])
+
+    return ret_action
+
+
+# NOTE: we need a type:ignore because we can't specify the type of objects
+# because it would be dependent on an optional igibson import which won't
+# get imported on CI
+def navigate_to_param_sampler(  # type: ignore
+        rng: Generator, objects) -> Array:
+    """Sampler for navigateTo option."""
+    assert len(objects) in [2, 3]
+    # The navigation nsrts are designed such that this is true (the target
+    # obj is always last in the params list).
+    obj_to_sample_near = objects[-1]
+    closeness_limit = max(
+        [
+            1.5,
+            np.linalg.norm(np.array(\
+                obj_to_sample_near.bounding_box[:2])) + 0.5,  # type: ignore
+        ]
+    )
+    distance = (closeness_limit - 0.01) * rng.random() + 0.03  # type: ignore
```

Comment:
can we head off this type: ignore and the one below that uses `distance` by doing `closeness_limit: float = max(...`?